imap_processing/glows/l1a/glows_l1a.py

```diff
@@ -88,6 +98,36 @@ def glows_l1a(packet_filepath: Path, data_version: str) -> list[xr.Dataset]:
     return output_datasets
 
 
+def determine_observational_day(hist_l0: list[HistogramL0]) -> list:
+    """
+    Find the timestamps for each observational day.
+
+    This function temporarily uses the is_night flag to determine the start of a new
+    observational day, but should eventually use the spin table APIs.
+
+    Parameters
+    ----------
+    hist_l0 : list[HistogramL0]
+        List of HistogramL0 objects.
+
+    Returns
+    -------
+    list
+        List of start times for each observational day.
+    """
+    prev_is_night = -1
+    obs_day_change = []
+    for hist in hist_l0:
+        flags = HistogramL1B.deserialize_flags(hist.FLAGS)
+        is_night = flags[6]
```

Comment:
Mypy seems to be flagging `prev_is_night` as an `integer,` based on the -1 assignment to it. But `is_night` is being found to be an `ndarray` based on where you are getting the value from which is `flags` which is an `ndarry.` I think by adding a simple type hint to the line should fix the issue assuming both variable should be interpreted as integers.

```suggestion
        is_night: int  = int(flags[6])
```